packages/amplify_core/lib/src/types/api/rest/http_payload.dart

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:async/async.dart';
+
+/// {@template amplify_core.http_payload}
+/// An HTTP request's payload.
+/// {@endtemplate}
+class HttpPayload extends StreamView<List<int>> {
+  /// {@macro amplify_core.http_payload}
+  factory HttpPayload([Object? body]) {
```

Comment:
Let's add one for form values like `package:http` provides

Reply:
Took a shot at this in next rev, lmk if needs more changes. Also, I'm assuming the caller would want to supply encoding header themselves when they do this? In package:http there is a check of the encoding type from the headers related to this. However, bc this is outside the request itself, not easy to replicate.

Reply:
Would try to figure out how to do it so that we set the correct content type header for them

Reply:
tried doing that in next rev